.github/workflows/steam.yml

```diff
@@ -0,0 +1,214 @@
+name: Steam Upload
+
+on:
+  release:
+    types:
+    - published
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to fetch and upload (nightly if none)'
+        required: false
+      win_url_override:
+        description: 'Windows build to use (.zip only)'
+        required: false
+      mac_url_override:
+        description: 'Mac build to use (.dmg only)'
+        required: false
+  schedule:
+  - cron: 0 0 * * *
+
+env:
+  WORKFLOW_ID: 583765
+  GIT_NIGHTLY_BRANCH: master
+  STEAM_NIGHTLY_BRANCH: nightly
+  STEAM_STABLE_BRANCH: staging
+  STEAM_BETA_BRANCH: beta_staging
+
+jobs:
+  upload:
+    name: Steam upload
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: source
+
+    - name: Setup 7zip
+      run: |
+        mkdir 7z
+        (
+            cd 7z
+            curl -s https://www.7-zip.org/a/7z2107-linux-x64.tar.xz | tar -xJ
```

Comment:
Any reason why you don't use `apt` to install 7z? Also might be worthwhile checking for a pre-existing `7z` on the runner to skip this step.

Reply:
7z doesn't seem to be listed in the ubuntu-20.04 image.

https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md

Reply:
p7zip is unmaintained and based on an ancient 7z version, so it doesn't support many newer formats like HFS+

Reply:
We could use this PPA instead: https://launchpad.net/~spvkgn/+archive/ubuntu/sevenzip
It has a recent enough version that would work.